pkg/build/api/types.go

```diff
@@ -707,6 +771,28 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the build request.
+	BuildCauses []*BuildTriggerCauses
```

Comment:
it really isn't. my reasoning was that this was a "lighter" version that just carried minimal information via the buildrequest from the controllers so that i don't attach the whole buildtriggerinfo struct and carry it over to the build instantiatior with just a few fields. come to think of it, duplication is probably worse :) should i make it a slice of pointers to buildtriggerinfo to take into account future level triggered builds, or just a pointer to that struct?